### PR TITLE
Fix chain validation

### PR DESCRIPTION
**Pull request by elmattic**
**Summary of changes**
Changes introduced in this pull request:
- Fix chain validation function

When we changed `FIL_PROOFS_PARAMETER_CACHE` from its default value (`/var/tmp/filecoin-proof-parameters`) to a configuration driven one, we have created a functional regression that stayed unnoticed for a while.

The regression shows up in [validate_chain](https://github.com/ChainSafe/forest/blob/main/blockchain/state_manager/src/lib.rs) function, resulting in an always broken state.

This PR fixes this issue by moving the init of proof parameters *before* `sync_from_snapshot` (that can call `validate_chain` internally), ensuring a proper init.

**Reference issue to close (if applicable)**

Closes n/a

**Other information and links**

### CODE REVIEW

File: forest/daemon/src/daemon.rs [hunk 1]
```diff
@@ -342,6 +342,18 @@ pub(super) async fn start(config: Config, detached: bool) {
...
+        use forest_paramfetch::{
+            get_params_default, set_proofs_parameter_cache_dir_env, SectorSizeOpt,
+        };
+        set_proofs_parameter_cache_dir_env(&config.client.data_dir);
+
+        get_params_default(&config.client.data_dir, SectorSizeOpt::Keys)
+            .await
+            .unwrap();
+    }
+
```
Comment (LesnyRumcajs) [inline forest/daemon/src/daemon.rs:356]: It would be nice to put it into a separate method. Not a big deal though.

**Comment by lemmih** [comment]
Oh my. I guess this wasn't caught because most of us still have a copy of the parameters at their default location.

**Review by LesnyRumcajs — APPROVED** [review]

**Comment by elmattic** [comment]
> Oh my. I guess this wasn't caught because most of us still have a copy of the parameters at their default location.

Exactly.

**Comment by LesnyRumcajs** [comment]
> Oh my. I guess this wasn't caught because most of us still have a copy of the parameters at their default location.

How did it work then on sync checks? I mean, the containers were always spawned with a fresh state, no?

**Review by hanabi1224 — APPROVED** [review]

**Review by lemmih — APPROVED** [review]

**Pull request merged**